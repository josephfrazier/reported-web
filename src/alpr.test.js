/**
 * @jest-environment node
 */
/* eslint-env jest */

import { readFileSync } from 'fs';

import readLicenseViaALPR from './alpr.js';

describe('readLicenseViaALPR', () => {
  test('using only the first token', async () => {
    const attachmentBuffer = readFileSync(
      './src/4d81b7e083843d949a73cc1178227fe8_photoData0.jpg',
    );
    const { PLATERECOGNIZER_TOKEN } = process.env;

    const result = await readLicenseViaALPR({
      attachmentBuffer,
      PLATERECOGNIZER_TOKEN,
    });

    expect(result.results[0]).toMatchSnapshot();
  });

  test('falling back to the second token', async () => {
    const attachmentBuffer = readFileSync(
      './src/4d81b7e083843d949a73cc1178227fe8_photoData0.jpg',
    );
    const { PLATERECOGNIZER_TOKEN_TWO } = process.env;

    const result = await readLicenseViaALPR({
      attachmentBuffer,
      PLATERECOGNIZER_TOKEN: 'INVALID TOKEN',
      PLATERECOGNIZER_TOKEN_TWO,
    });

    expect(result.results[0]).toMatchSnapshot();
  });
});
