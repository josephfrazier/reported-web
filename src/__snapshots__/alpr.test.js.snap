// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readLicenseViaALPR falling back to the second token 1`] = `
{
  "box": {
    "xmax": 1842,
    "xmin": 1638,
    "ymax": 2112,
    "ymin": 2009,
  },
  "candidates": [
    {
      "plate": "t736394c",
      "score": 0.9,
    },
  ],
  "dscore": 0.729,
  "plate": "t736394c",
  "region": {
    "code": "us-ny",
    "score": 0.859,
  },
  "score": 0.9,
  "vehicle": {
    "box": {
      "xmax": 2441,
      "xmin": 1260,
      "ymax": 2236,
      "ymin": 1176,
    },
    "score": 0.89,
    "type": "Sedan",
  },
}
`;

exports[`readLicenseViaALPR using only the first token 1`] = `
{
  "box": {
    "xmax": 1842,
    "xmin": 1638,
    "ymax": 2112,
    "ymin": 2009,
  },
  "candidates": [
    {
      "plate": "t736394c",
      "score": 0.9,
    },
  ],
  "dscore": 0.729,
  "plate": "t736394c",
  "region": {
    "code": "us-ny",
    "score": 0.859,
  },
  "score": 0.9,
  "vehicle": {
    "box": {
      "xmax": 2441,
      "xmin": 1260,
      "ymax": 2236,
      "ymin": 1176,
    },
    "score": 0.89,
    "type": "Sedan",
  },
}
`;
